torch/csrc/autograd/engine.cpp

```diff
@@ -75,6 +76,25 @@ inline bool should_run_in_cpu_ready_queue(c10::DeviceType device) {
     return false;
   }
 }
+
+static std::atomic<Engine::compiled_autograd_fn> the_compiled_autograd =
+    nullptr;
+static std::atomic<int32_t> num_threads_in_backwards;
+struct CompiledAutogradThreadingDebugCheck {
+  CompiledAutogradThreadingDebugCheck() : active(true) {
+    num_threads_in_backwards++;
+  }
+  ~CompiledAutogradThreadingDebugCheck() {
+    release();
+  }
+  void release() {
+    if (std::exchange(active, false)) {
```

Comment:
Why is this not decrement the counter and if it reached zero flip the active flag?

Reply:
The active flag has nothing to do with the counter being zero.  The active flag is just checking if release() has already been called to prevent a double decrement.

Reply:
Ho ok. I guess the whole active thing is not necessary since it's only used if compiled autograd is already active? Or it is also used to disable it while it is running and so you need this mechanism?

Reply:
It is to prevent:
```
x.release();  // (called directly below in an `if()`)
x.release();  // (indirectly in destructor)
```
from decrementing the counter twice.  

It has nothing to do with compiled autograd being active.  It is just a simple bit to check if we already decremented the counter.

There is a single call to `release()` below (to allow `compled_autograd.disable()` in a nested call), that without this would result in a double decrement when the object is destructed.

Reply:
I renamed this to `incremented` to hopefully be more clear.